### PR TITLE
Arturo

### LINKED ISSUE #2: Create Event
as a `partner` I want to create a new event

**Input Data**:
- event name
- start date
- expiration date
- start hour
- expiration hour
- poc chuc torta unit price
- shrimp torta unit price

**Contraints**:
- event name is required and a string
- start date must be a future date and an epoch (dto)
- expiration date must be a future date, must be after start date and and epoch (dto)
- start hour must be epoch
- expiration hour must epoch
- poc chuc torta unit price must be a non negative number
- shrimp torta unit price must be a non negative number
- partner must be logged in

---

**Scenario 1**
user tries to create new user with the following data

- event name: Tortas app
- start date: today
- end date: tomorrow
- start hour: 10am
- end hour: 11am
- poc chuc torta unitary price: 10
- camaron torta unitary price: 12

*System Response*: notifies the user that the event was created

---

**Scenario 2**: 

the user tries to create a new event but inputs the wrong data,

*System Response*: display error messages below the fields (see error list)

****Error List****

- event name
  - empty name: "Is required"
- start date:
  - empty: "Is required"
  - before current date: "Must be a future date"
  - invalid format: "Must be epoch time"
- expiration date:
  - empty: "Is required"
  - before start date: "Must be after start date"
  - invalid format: "Must be epoch time"
- start hour:
  - empty: "Is required"
  - not in the range (from 0 to 24 * 60): "Must be in the range from 0 to 24 * 60"
  - invalid format: "Must be a number"
- expiration hour:
  - empty: "Is required"
  - before start hour: "Must be after start hour"
  - not in the range (from 0 to 24 * 60): "Must be in the range from 0 to 24 * 60"
  - invalid format: "Must be a number"
- poc chuc torta unit price
  - empty: "Is required"
  - invalid format: "Must be a non-negative number"
- shrimp torta unit price
  - empty: "Is required"
  - invalid format: "Must be a non-negative number"

### PR DESCRIPTION
**Pull request by CanKer**
Closes #2

### CODE REVIEW

File: src/components/Order.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import styled from "@emotion/styled";
+
+import { IOrderEntity } from "./../interfaces/";
+
+interface IProps {
+  orders: IOrderEntity[];
+}
+
+export const Order: React.SFC<IProps> = props => {
+  // const DivOrder = styled.div({
```
Comment (mishelashala) [inline src/components/Order.tsx]: delete commented/unused code

File: src/components/event/Create.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  public startDate = React.createRef<HTMLInputElement>();
+  public endDate = React.createRef<HTMLInputElement>();
+  public startHour = React.createRef<HTMLInputElement>();
+  public endHour = React.createRef<HTMLInputElement>();
+  public tortaPocchuc = React.createRef<HTMLInputElement>();
+  public tortaCamaron = React.createRef<HTMLInputElement>();
+
+  public createEvent = (e: any) => {
+    e.preventDefault();
+    const validation =
```
Comment (mishelashala) [inline src/components/event/Create.tsx]: extract logic into a function

File: src/components/event/Create.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+export class CreateEvent extends React.Component {
+  public state = {
+    event: {}
+  };
+
+
+
+  public eventName = React.createRef<HTMLInputElement>();
```
Comment (mishelashala) [inline src/components/event/Create.tsx]: why do yo need the refs?

File: src/components/orders/DataOrder [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
```
Comment (mishelashala) [inline src/components/orders/DataOrder]: data order has no extension file

**Review by mishelashala — CHANGES_REQUESTED** [review]

File: src/components/orders/DataOrder (hunk 4 above)
Comment (CanKer) [reply src/components/orders/DataOrder]: Yeah, I'll untrack

File: src/components/Order.tsx (hunk 1 above)
Comment (CanKer) [reply src/components/Order.tsx]: untracking

File: src/components/event/Create.tsx (hunk 3 above)
Comment (CanKer) [reply src/components/event/Create.tsx]: for the data in the form

**Review by mishelashala — APPROVED** [review]

**Pull request merged**